src/app/plan/components/Wizard/HooksFormComponent.tsx

```diff
@@ -0,0 +1,442 @@
+import React, { useState, useEffect } from 'react';
+import { FormikProps } from 'formik';
+import {
+  Grid,
+  GridItem,
+  Text,
+  TextContent,
+  TextVariants,
+  FormGroup,
+  Radio,
+  TextInput,
+  Button,
+  FileUpload,
+  Tooltip,
+  TooltipPosition,
+  Form
+} from '@patternfly/react-core';
+import spacing from '@patternfly/react-styles/css/utilities/Spacing/spacing';
+import {
+  AddEditMode,
+  addEditButtonText,
+} from '../../../common/add_edit_state';
+import { OutlinedQuestionCircleIcon } from '@patternfly/react-icons';
+import SimpleSelect from '../../../common/components/SimpleSelect';
+const classNames = require('classnames');
+
+const componentTypeStr = 'hook';
+const addEditButtonTextFn = addEditButtonText(componentTypeStr);
+const styles = require('./HooksForm.module');
+
+interface IFormValues {
+  hookName: string;
+  hookImageType: string;
+  ansibleFile: string;
+  ansibleFilename: string;
+  ansibleRuntimeImage: string;
+  customContainerImage: string;
+  srcServiceAccountName: string;
+  srcServiceAccountNamespace: string;
+  destServiceAccountName: string;
+  destServiceAccountNamespace: string;
+  clusterType: string;
+  migrationStep: string;
+}
+interface IOtherProps {
+  onAddEditSubmit: any;
+  onClose: any;
+  isEdit: boolean;
+  isEditHook: any;
+  setInitialHookValues: any;
+  setIsAddHooksOpen: any;
+  hookAddEditStatus: any;
+  cancelAddEditWatch: () => void;
+  resetAddEditState: () => void;
+  currentPlan: any;
+}
+
+const HooksFormComponent = (props: IOtherProps & FormikProps<IFormValues>) => {
+
+  const {
+    setIsAddHooksOpen,
+    setInitialHookValues,
+    hookAddEditStatus,
+    handleSubmit,
+    touched,
+    errors,
+    setFieldValue,
+    setFieldTouched,
+    values,
+    handleChange,
+    handleBlur,
+    cancelAddEditWatch,
+    resetAddEditState,
+    currentPlan
+  } = props;
+
+  const formikHandleChange = (_val, e) => handleChange(e);
+  const formikSetFieldTouched = key => () => setFieldTouched(key, true, true);
+
+  const hookNameKey = 'hookName';
```

Comment:
I think it makes sense to leave these here. Otherwise we are risking typos when retyping the string for fieldId, name, and anywhere it is passed in as a param to the formik value change wrapper functions. I vote to keep them for now. This is definitely up for discussion, though. There might be a way to clean all of this up including the formik wrapper functions. Not sure if those are still needed.

Reply:
> I think it makes sense to leave these here. Otherwise we are risking typos when retyping the string for fieldId, name, and anywhere it is passed in as a param to the formik value change wrapper functions.

+1, this is why I defined these as consts. Always prefer const defs for hooks rather than string literals that can be botched. Unless I'm misunderstanding some way that typescript will prevent this.

Reply:
I'm fine with that, especially because I'm not 100% sure the formik typings support this. But in general if a function argument has a type that is an enum or a specific set of values (e.g. `foo: 'a' | 'b' | 'c'`) TypeScript will catch a string literal that doesn't match, and IntelliSense will autocomplete string literals from the valid values. I would hope formik uses the `IFormValues` keys to populate that argument's type. Let's leave this though, nothing wrong with a little more confidence.